src/CompressedWriter.cc

```diff
@@ -61,7 +61,18 @@ CompressedWriter::CompressedWriter(const string& filename, size_t block_size,
   // until we've finished initializing it.
   pthread_mutex_lock(&mutex);
   for (uint32_t i = 0; i < num_threads; ++i) {
-    pthread_create(&threads[i], nullptr, compression_thread_callback, this);
+    while (true) {
+      int err = pthread_create(&threads[i], nullptr, compression_thread_callback, this);
+      if (err == EAGAIN) {
+        sched_yield(); // Give other processes a chance to exit.
+        continue;
+      } else if (err != 0) {
+        static char errmsg[] = "FATAL: Failed to create compression threads.\n";
+        ::write(STDERR_FILENO, errmsg, sizeof(errmsg));
+        abort();
```

Comment:
Why aren't we calling rr's FATAL() here?

Reply:
log.h is not included in this file, because it can't be used from threads, so I didn't want to change that to avoid somebody accidentally introducing a log statement somewhere other than the main thread in the future.

Reply:
I think we should at least print the error code.